proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,452 @@
+- **Internals mechanism** - The React Native package is responsible of encapsulating away complexity and handling mechanisms such as:
+  - The threading model
+  - Interoperability layers between different architectures
+- **JS Engine Integration** - The React Native package is responsible for offering an API to integrate a target JS engine. React Native is distributed with built-in integrations for Hermes and JavaScriptCore. Third-party engines (like V8) may be integrated but are not directly supported.
+- **Bundler integration** - The React Native package defines the integration with Metro, the React Native bundler. Part of this API surface are:
+  - Building the JS and asset bundle
+  - The development server
+  - Debugging integrations
+  - Sourcemap support
+  - Error and Crash reporting
+- **Native build integration** - The React Native package is responsible for defining how the Android & iOS final artifact are built, as well as how Java/Kotlin/Swift/ObjC/ObjC++ and C++ should be compiled and distributed.
+- **3rd party libraries** - The React Native package is responsible for defining integration API for 3rd party libraries (i.e. how to create libraries for React Native and how to integrate them with the runtime). Frameworks can build on top of those APIs to simplify the library integrations and discovery via mechanisms such as auto-linking.
+
+### What does the React Native package not do
+
+React Native Frameworks provide features that **most native apps** that are intending to reach production eventually need.
+
+Here is a (non-exhaustive) list of areas of what the React Native package is **not directly focusing** on, and is delegating this responsibility to frameworks:
+
+- **Project Scaffolding** - Frameworks can offer tools and templates to create new projects so that users can get started developing in a matter of seconds.
+- **Tooling for React Native upgrade DevX** - Similarly to project scaffolding, framework can offer tools to make sure the project can easily keep up with the latest React Native version. Tools such as the upgrade helper, Expo Config Plugins and others fit in this category.
+- **Navigation and Routing solutions** - React Native doesn’t offer a navigation or a routing solution out of the box. Frameworks can either offer their own solution or recommend a preferred navigation solution to use.
+- **Managing the build experience** - Framework can offer tools to simplify the native builds and move to fully managed native builds that are happening in the cloud.
+- **Manage the publishing experience** - Frameworks can offer solutions to publish applications to public stores (App Store/Play Store and others) either locally or managed in the cloud.
+- **Providing OTA updates** - Frameworks can offer mechanisms to deploy Over-the-air updates to users. Tools such as EAS Update & Codepush fall into this category.
```

Comment:
Maybe the framework should provide an interface for any third party to develop their own OTA update functionality, so developer could choose whatever provider they want but to ensure the basic OTA functionality work the same across all different providers

Reply:
As @brentvatne mentioned in [this other comment](https://github.com/react-native-community/discussions-and-proposals/pull/759#discussion_r1476519803), yes we'd love to have more formal APIs and RFCs a around those APIs.

Historically, we never really provided clear boundaries/APIs for framework developers, OTA updates is one of this scenario but not the only one.
As the React Native matures, we'll be providing more precise API boundaries and documentation for developers.

> so developer could choose whatever provider they want but to ensure the basic OTA functionality work the same across all different providers

Yup, React Native will be providing the core functionalities, and Frameworks can customize them by offering top-notch developer experience with their own flavor.